src/theme.c

```diff
@@ -1042,3 +1044,40 @@ theme_finish(struct theme *theme)
 	zdrop(&theme->corner_top_right_active_normal);
 	zdrop(&theme->corner_top_right_inactive_normal);
 }
+
+struct theme
+get_theme_for_view(struct view *view)
+{
+	float *custom_color = window_rules_get_custom_border_color(view);
+	if (!custom_color) {
+		return *view->server->theme;
+	}
+
+	struct theme theme = { 0 };
```

Comment:
Declaring `theme` here and then returning it, doesn't look right. Its life ends with this frame.